### PR TITLE
Calculate trigram auto-brightness in worker thread

### PR DESCRIPTION
**Pull request by MaciekPytel**
Added functionality to base VisualisationWidget to make
this sort of async pre-calculation easy.

### CODE REVIEW

File: src/visualisation/trigram.cc [hunk 1]
```diff
@@ -212,6 +220,15 @@ int TrigramWidget::suggestBrightness() {
                   k_brightness_heuristic_max - offset);
 }
 
+VisualisationWidget::AdditionalResampleData* TrigramWidget::onAsyncResample() {
+  if (use_brightness_heuristic_) {
+    BrightnessData* res = new BrightnessData();
```
Comment (adamiwaniuk) [inline src/visualisation/trigram.cc]: where this data is deleted ?

Comment (MaciekPytel) [inline src/visualisation/trigram.cc]: It's later inserted into shared_ptr (base.cc, line 107), which will handle deletion.

File: include/visualisation/base.h [hunk 2]
```diff
@@ -43,22 +44,40 @@ class VisualisationWidget : public QOpenGLWidget,
   // options of this visualisation. Return true if anything was added to
   // QLayout.
   virtual bool prepareOptionsPanel(QBoxLayout *layout);
-  void refreshVisualisation();
+
+  /**
+   * Derive this if you want to produce some additional data in
+   * onAsyncResample method.
+   */
+  struct AdditionalResampleData {};
```
Comment (koriakin) [inline include/visualisation/base.h]: Wirtualny destruktor?

Comment (MaciekPytel) [inline include/visualisation/base.h]: fixed

**Pull request merged**